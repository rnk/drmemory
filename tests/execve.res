# **********************************************************
# Copyright (c) 2009-2010 VMware, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; 
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
Error #1: UNADDRESSABLE ACCESS: reading 1 byte(s)
malloc.c:95
Error #2: UNINITIALIZED READ
malloc.c:105
Error #3: UNINITIALIZED READ
malloc.c:118
Error #4: INVALID FREE
!if WINDOWS
# addr2line and winsyms report slightly different results here
malloc.c:160
!endif
!if UNIX
malloc.c:161
!endif
!if UNIX
# on unix free touches the invalid address; not so on winxp it seems
Error #5: UNADDRESSABLE ACCESS: reading 4 byte(s)
malloc.c:161
!endif
