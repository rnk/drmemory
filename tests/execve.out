# **********************************************************
# Copyright (c) 2009-2010 VMware, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; 
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
parent is running
child is running
malloc
malloc small
malloc 0
malloc big
calloc
realloc
invalid free
all done
child has exited
:::Dr.Memory::: ERRORS FOUND:
# on unix free touches the invalid address; not so on winxp it seems
!if WINDOWS
:::Dr.Memory:::       1 unique,    20 total unaddressable access(es)
!endif
!if UNIX
:::Dr.Memory:::       2 unique,    21 total unaddressable access(es)
!endif
:::Dr.Memory:::       2 unique,     2 total uninitialized access(es)
:::Dr.Memory:::       1 unique,     1 total invalid free(s)
:::Dr.Memory:::       0 unique,     0 total warning(s)
:::Dr.Memory:::       2 total,     59 byte(s) of leak(s)
:::Dr.Memory:::       0 total,      0 byte(s) of possible leak(s)
